tools/python_api/src_py/torch_geometric_result_converter.py

```diff
@@ -0,0 +1,237 @@
+from .types import Type
+import warnings
+
+
+class TorchGeometricResultConverter:
+    def __init__(self, query_result):
+        self.query_result = query_result
+        self.nodes_dict = {}
+        self.edges_dict = {}
+        self.rels = set()
+        self.nodes_property_names_dict = {}
+        self.table_to_label_dict = {}
+        self.internal_id_to_pos_dict = {}
+        self.pos_to_primary_key_dict = {}
+        self.warning_messages = set()
+        self.ignored_columns = set()
+        self.column_to_extract = self.query_result._get_columns_to_extract()
+
+    def __get_node_property_names(self, table_name):
+        if table_name in self.nodes_property_names_dict:
+            return self.nodes_property_names_dict[table_name]
+
+        PRIMARY_KEY_SYMBOL = "(PRIMARY KEY)"
+        LIST_SYMBOL = "[]"
+        result_str = self.query_result.connection._connection.get_node_property_names(
```

Comment:
This comment is beyond this PR, but let me document it here:

The parsing here makes me think that our cpp interface `string Connection::getNodePropertyNames(const string& tableName)` is incorrect. Instead of returning a `string`, it's better to return `vector<string>`, each inside which is a property name. The caller (shell) shall be responsible to organize them in a printable way.